src/stores/absence.js

```diff
@@ -0,0 +1,55 @@
+/**
+ * @copyright Copyright (c) 2023 Maksim Sukharev <antreesy.web@gmail.com>
+ *
+ * @author Maksim Sukharev <antreesy.web@gmail.com>
+ *
+ * @license AGPL-3.0-or-later
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+import { defineStore } from 'pinia'
+import Vue from 'vue'
+
+import { getUserAbsence } from '../services/participantsService.js'
+
+export const useAbsenceStore = defineStore('absence', {
+	state: () => ({
+		absence: {},
+	}),
+
+	actions: {
+		/**
+		 * Fetch an absence status for user and save to store
+		 *
+		 * @param {object} payload action payload
+		 * @param {string} payload.token The conversation token
+		 * @param {string} payload.userId The id of user
+		 *
+		 */
+		async getUserAbsence({ token, userId }) {
+			if (this.absence[token]) {
+				return this.absence[token]
+			}
+
+			try {
+				const response = await getUserAbsence(userId)
+				Vue.set(this.absence, token, response.data.ocs.data)
```

Comment:
If so,  don't forget this
```suggestion
				Vue.set(this.absence, token, response.data.ocs.data)
				return this.absence[token]
```